app/gulpfile.js

```diff
@@ -0,0 +1,109 @@
+var path = require('path');
+var del = require('del');
+var gulp = require('gulp'); //Gulp 4.0+
+var webpack = require('gulp-webpack');
+var ts = require('gulp-typescript');
+var typescript = require('typescript');
+var sourcemaps = require('gulp-sourcemaps');
+var mochaPhantomJS = require('gulp-mocha-phantomjs');
+var istanbul = require('gulp-istanbul');
+var istanbulReport = require('gulp-istanbul-report');
+var gls = require('gulp-live-server');
+var gulpFilter = require('gulp-filter');
+var lint = require('gulp-tslint');
+var gutil = require('gulp-util');
+var tslint = require('tslint');
+
+// joins paths to represent a file relative to the current directory
+var _localFile = function(file) {
+  return path.join(__dirname, file);
+}
+
+gulp.task('build:app', gulp.series(function() {
+  var project = ts.createProject(_localFile('tsconfig.json'), {typescript: typescript});
+
+  return project.src()
+    .pipe(sourcemaps.init())
+    .pipe(ts(project))
+    .pipe(sourcemaps.write())
+    .pipe(gulp.dest(_localFile('tmp'))); // Write every JS file to tmp/
+}, function() {
```

Comment:
Why are we writing the js files to disk at all? Gulp uses a virtual filesystem to track this stuff. Shouldn't we just pipe it into the next step at line 32?